src/flyte/extend.py

```diff
@@ -1,14 +1,17 @@
 from ._initialize import is_initialized
+from ._internal.imagebuild.image_builder import ImageBuildEngine
 from ._internal.runtime.entrypoints import download_code_bundle
 from ._internal.runtime.resources_serde import get_proto_resources
 from ._resources import PRIMARY_CONTAINER_DEFAULT_NAME, pod_spec_from_resources
-from ._task import AsyncFunctionTaskTemplate
+from ._task import AsyncFunctionTaskTemplate, TaskTemplate
 from ._task_plugins import TaskPluginRegistry
 
 __all__ = [
     "PRIMARY_CONTAINER_DEFAULT_NAME",
     "AsyncFunctionTaskTemplate",
+    "ImageBuildEngine",
```

Comment:
should this be here?  or in the main init file?  `build_images` and `Image` are already exposed there.  I think we should keep these together.  Also does accessing ImageBuildEngine do something that build_images can't do?